### PR TITLE
Master command smart action fge

### PR DESCRIPTION
**Pull request by FrancoisGe**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/legacy/js/fields/relational_fields.js [hunk 1]
```diff
@@ -1025,6 +1025,51 @@ var KanbanFieldMany2One = AbstractField.extend({
 const Many2OneAvatar = FieldMany2One.extend({
     _template: 'web.Many2OneAvatar',
 
+    on_attach_callback() {
+        const self = this;
+        if(self.viewType === "form") {
+            core.bus.trigger("set_legacy_command", (env) => ({
```
Comment (brboi) [inline addons/web/static/src/legacy/js/fields/relational_fields.js:1031]:
```suggestion
            core.bus.trigger("set_legacy_command", "web.Many2OneAvatar.assignTo", (env) => ({
```

File: addons/web/static/src/legacy/utils.js [hunk 2]
```diff
@@ -120,6 +120,40 @@ export function makeLegacyDialogMappingService(legacyEnv) {
...
+ */
+export function makeLegacyCommandService(legacyEnv) {
+    return {
+        dependencies: ["command"],
+        start(env) {
+            const { command } = env.services;
+
+            const commandRemoveMap = new Map();
+
+            function setLegacyCommand(data) {
```
Comment (brboi) [inline addons/web/static/src/legacy/utils.js:137]:
```suggestion
            function setLegacyCommand(uniqueId, getCommandDefinition) {
```

File: addons/web/static/src/legacy/js/fields/relational_fields.js [hunk 3]
```diff
@@ -2757,6 +2833,70 @@ var FieldMany2ManyTagsAvatar = FieldMany2ManyTags.extend({
...
+                            ids: { id:  session.user_id[0], display_name: session.name }
+                        });
+                    }
+                },
+            });
+            core.bus.trigger("set_legacy_command", "web.Many2OneAvatar.assignToMe", getCommandDefinition);
+        }
+    },
+    on_detach_callback() {
+        core.bus.trigger("remove_legacy_command", "web.FieldMany2ManyTagsAvatar.assignTo");
```
Comment (kebeclibre) [inline addons/web/static/src/legacy/js/fields/relational_fields.js:2896]: Note that for this widget specifically, the relevance of having multiple widgets of that type in one view is very high, so having on the last widget's command active bothers me. I guess we'd need to improve this in the future

**Review by seb-odoo — CHANGES_REQUESTED** [review]
I was asked to review this but this is 100% web client code, I don't really have much to say about the diff that is currently in `mail` module.

Just that probably it should not be in `mail` module.

I think the only reason the field avatar and such were in `mail` is because they were original made just to open a chat on click.

So my only suggestion would be to move all of this code in `web` and only leave a very small override in `mail` just for the open chat part.

**Comment by kebeclibre** [comment]
@seb-odoo Heardya'h :D I'm gonna merge to avoid missing any deadline, and create a task to do the small refactor you ask for.
Thanks for reviewing

robodoo r+

File: addons/mail/static/src/js/m2x_avatar_user.js [hunk 4]
```diff
@@ -58,6 +59,124 @@ export const Many2OneAvatarUser = Many2OneAvatar.extend(M2XAvatarMixin, {
         'click .o_m2o_avatar > img': '_onAvatarClicked',
     }),
 
+    on_attach_callback() {
+        console.log("on_attach_callback ", this.mode)
```
Comment (brboi) [inline addons/mail/static/src/js/m2x_avatar_user.js:63]: oups

File: addons/web/static/src/legacy/js/fields/basic_fields.js [hunk 5]
```diff
@@ -2504,6 +2504,44 @@ var PriorityWidget = AbstractField.extend({
         var checked = this.$("[aria-checked='true']");
         return checked.length ? checked : this.$("[data-index='1']");
     },
+    init: function () {
+        this._super.apply(this, arguments);
+        console.log("init ", this.mode)
```
Comment (brboi) [inline addons/web/static/src/legacy/js/fields/basic_fields.js:2509]: oups

File: addons/web/static/src/legacy/js/fields/basic_fields.js [hunk 6]
```diff
@@ -2504,6 +2504,44 @@ var PriorityWidget = AbstractField.extend({
...
+    },
+
+    on_attach_callback() {
+        console.log("on_attach_callback ")
```
Comment (brboi) [inline addons/web/static/src/legacy/js/fields/basic_fields.js:2513]: oups

File: addons/web/static/src/legacy/utils.js [hunk 7]
```diff
@@ -120,6 +120,41 @@ export function makeLegacyDialogMappingService(legacyEnv) {
...
+export function makeLegacyCommandService(legacyEnv) {
+    return {
+        dependencies: ["command"],
+        start(env) {
+            const { command } = env.services;
+
+            const commandRemoveMap = new Map();
+
+            function setLegacyCommand(uniqueId, getCommandDefinition) {
+                console.log("setLegacyCommand ", uniqueId);
```
Comment (brboi) [inline addons/web/static/src/legacy/utils.js:138]: oups

**Pull request closed without merging**